Sources/Starknet/Network/HttpNetworkProvider.swift

```diff
@@ -47,16 +47,30 @@ class HttpNetworkProvider {
         return request
     }
 
+    private func handleEncodingError(_ error: Error) throws -> Never {
+        if let encodingError = error as? EncodingError {
+            throw HttpNetworkProviderError.encodingError(encodingError)
+        } else {
+            throw HttpNetworkProviderError.unknownError
+        }
+    }
+
+    private func handleResponseError(_ error: Error, _ response: URLResponse) throws -> Never {
+        if let response = response as? HTTPURLResponse, response.statusCode < 200 || response.statusCode > 299 {
+            throw HttpNetworkProviderError.requestRejected
+        } else if let decodingError = error as? DecodingError {
+            throw HttpNetworkProviderError.decodingError(decodingError)
+        } else {
+            throw HttpNetworkProviderError.unknownError
+        }
+    }
+
     func send<U>(payload: some Encodable, config: Configuration, receive _: U.Type) async throws -> U where U: Decodable {
         let encoded: Data
         do {
             encoded = try JSONEncoder().encode(payload)
         } catch {
-            if let encodingError = error as? EncodingError {
-                throw HttpNetworkProviderError.encodingError(encodingError)
-            } else {
-                throw HttpNetworkProviderError.unknownError
-            }
+            try handleEncodingError(error)
         }
```

Comment:
I also had that on my mind, but consider that `payload`  has different type in `send` methods. Because of that we would have to create two separate methods accepting `some Encodable` and `[some Encodable]`. Do you think it's worth to create them in this situation?

Reply:
Can't we just use `Any: Encodable`?

Reply:
Fine, we can 👍